baserun/baserun.py

```diff
@@ -20,245 +48,326 @@ class BaserunEvaluationFailedException(Exception):
+
+        processor = processor_class(BaserunExporter())
+        tracer_provider.add_span_processor(processor)
+
+        if find_spec("openai") is not None:
+            from .instrumentation.openai import OpenAIInstrumentor
+
+            instrumentor = OpenAIInstrumentor()
+            if not instrumentor.is_instrumented_by_opentelemetry:
+                instrumentor.instrument()
+            Baserun._instrumentors.append(instrumentor)
+
+        if find_spec("anthropic") is not None:
+            from .instrumentation.anthropic import AnthropicInstrumentor
+
+            instrumentor = AnthropicInstrumentor()
+            if not instrumentor.is_instrumented_by_opentelemetry:
+                instrumentor.instrument()
+            Baserun._instrumentors.append(instrumentor)
 
     @staticmethod
-    def _finish_trace(trace_type: TraceType):
-        if trace_type == TraceType.PRODUCTION:
-            Baserun.flush()
+    def uninstrument():
+        for instrumentor in Baserun._instrumentors:
+            instrumentor.uninstrument()
+
+        Baserun._instrumentors = []
+
+    @staticmethod
+    def serialize_run(run: Run) -> str:
+        """Serialize a Run into a base64-encoded string"""
+        return b64encode(run.SerializeToString()).decode("utf-8")
+
+    @staticmethod
+    def deserialize_run(run_serialized: str) -> Union[Run, None]:
+        """Deserialize a base64-encoded string into a Run"""
+        if not run_serialized:
+            return None
+
+        decoded = b64decode(run_serialized.encode("utf-8"))
+        run = Run()
+        run.ParseFromString(decoded)
+        return run
 
-        Baserun._buffer = []
-        Baserun._evals = []
-        Baserun._trace_id = None
+    @staticmethod
+    def current_run() -> Union[Run, None]:
+        """Gets the current run"""
+        current_span: _Span = get_current_span()
+        if not current_span.is_recording():
+            return None
+
+        current_run_str = current_span.attributes.get(SpanAttributes.BASERUN_RUN)
+        if not current_run_str:
+            return None
+
+        return Baserun.deserialize_run(current_run_str)
+
+    @staticmethod
+    def clear_run():
+        attach(set_value(SpanAttributes.BASERUN_RUN, None))
+
+    @staticmethod
+    def get_or_create_current_run(
+        name: str = None,
+        suite_id: str = None,
+        start_timestamp: datetime = None,
+        completion_timestamp: datetime = None,
+        trace_type: Run.RunType = None,
+        metadata: dict[str, Any] = None,
+    ) -> Run:
+        """Gets the current run or creates one"""
+        existing_run = Baserun.current_run()
+        if existing_run:
+            return existing_run
+
+        run_id = str(uuid.uuid4())
+        if not trace_type:
+            trace_type = (
+                Run.RunType.RUN_TYPE_TEST
+                if Baserun.current_test_suite
+                else Run.RunType.RUN_TYPE_PRODUCTION
+            )
+
+        if not name:
```

Comment:
Just make this required in the function signature?

Reply:
Actually this was unused, deleting it

Reply:
Oh, turns out we need it. This is a "get or create" function and we can get the current run without a name, but to create one we need a name